packages/ts-plugin/src/index.ts

```diff
@@ -1,6 +1,33 @@
 import { createLanguageServicePlugin } from '@volar/typescript/lib/quickstart/createLanguageServicePlugin.js';
+import type { CollectionConfig } from './frontmatter.js';
+import { getFrontmatterLanguagePlugin } from './frontmatter.js';
 import { getLanguagePlugin } from './language.js';
+import type { LanguagePlugin } from '@volar/language-core';
 
-export = createLanguageServicePlugin(() => ({
-	languagePlugins: [getLanguagePlugin()],
-}));
+export = createLanguageServicePlugin((ts, info) => {
+	let collectionConfig = undefined;
+
+	try {
+		const currentDir = info.project.getCurrentDirectory();
+		const fileContent = ts.sys.readFile(currentDir + '/.astro/collections/collections.json');
+		if (fileContent) {
+			collectionConfig = {
+				folder: currentDir,
+				config: JSON.parse(fileContent) as CollectionConfig['config'],
+			};
+		}
+	} catch (err) {
+		// If the file doesn't exist, we don't really care, but if it's something else, we want to know
+		if (err && (err as any).code !== 'ENOENT') console.error(err);
+	}
+
+	let languagePlugins: LanguagePlugin<string>[] = [getLanguagePlugin()];
+
+	if (collectionConfig) {
+		languagePlugins.push(getFrontmatterLanguagePlugin([collectionConfig]));
+	}
```

Comment:
Unfortunately in TS plugins getting any kind of config (apart from `tsconfig.json`) is really annoying. As such, the feature is always enabled there when it finds a config.